### PR TITLE
feat: Add support for Teams Adaptive cards in QnA Dialog

### LINKED ISSUE #4476: QnAMakerDialog cannot render rich text in Teams channel
The following text is copied from an **Internal Ticket**:

The problem happens on the Azure Cognitive Service - Custom Question Answering exported Bot application.
 
[PROBLEM DESCRIPTION]
This is a UI rendering issue only happens on Teams channel.
I notice the CQA generated Bot application is including rich text (based on Markdown format), it can be rendered in Web Chat correctly. 
 
1. In a Web Chat bot, markdown formatting is rendered correctly.
 
2. In a Teams chat bot, the formatting is not rendered correctly.
 
I digest into the Bot application generated by CQA, seems it's based on the BotBuilder samples - QnAMaker bot. I use Nodejs SDK but .NET SDK shows the same problem. 
 
I tried to update all BotBuilder dependencies to latest 4.19.2 but still no improvements.
If this is a mis-usage of the QnAMakerDialog class, which class should be used to show the same layout as Web Chat?

### PR DESCRIPTION
**Pull request by anishprasad01**
Fixes Internal IcM Issue #373550043
Fixes #4476

## Description

This PR allows for choosing Adaptive Cards formatted for use in Microsoft Teams instead of the default Hero Cards. This is done to support markdown formatting in QnA and CQA responses since Teams only supports markdown in Adaptive Cards. Changes are made in `qnaCardBuilder.ts` and `qnaMakerDialog.ts`.

Using the new cards simple requires adding code in the bot to take in a new environment variable: `UseTeamsAdaptiveCard = "true"` and pass it to the QnA dialog constructor. If no changes are made, the bot will default to a Hero card.

## Specific Changes

  - Create new methods to generate adaptive and hero cards
  - Modify the card builder class to take in a new bool field to allow for choosing whether or not to use an adaptive card
  - Modify the dialog class to accept the aforementioned bool field and pass it to the card builder.

## Testing

Tests ran successfully.
Screenshot 2023-04-27 115625

### CODE REVIEW

File: libraries/botbuilder-ai/src/qnaCardBuilder.ts [hunk 1]
```diff
@@ -113,4 +119,63 @@ export class QnACardBuilder {
...
     }
+
+    /**
+     * Returns an Adaptive Card attachment containing the text for the card and a button list formatted for MS Teams
+     *
+     * @param {string} cardText The string to be placed in the card's text field
+     * @param {any[]} buttonList The list of buttons to be converted to MS Teams messageBack buttons and placed in the card's actions field
+     * @returns {Attachment} An attachment representing the MS Teams-formatted Adaptive Card
+     */
+    static getTeamsAdaptiveCard(cardText: string, buttonList: any[]) {
```
Comment (ceciliaavila) [inline libraries/botbuilder-ai/src/qnaCardBuilder.ts:130]:
```suggestion
    static getTeamsAdaptiveCard(cardText: string, buttonList: any[]): Attachment {
```

File: libraries/botbuilder-ai/src/qnaCardBuilder.ts [hunk 2]
```diff
@@ -113,4 +119,63 @@ export class QnACardBuilder {
...
+    }
+
+    /**
+     * Returns a Hero Card attachment containing the text for the card and a button list
+     *
+     * @param {string} cardText The string to be placed in the card's text field
+     * @param {any[]} buttonList The list of buttons to be converted to imBack buttons and attached to the card
+     * @returns {Attachment} An attachment representing the Hero Card
+     */
+    static getHeroCard(cardText: string, buttonList: any[]) {
```
Comment (ceciliaavila) [inline libraries/botbuilder-ai/src/qnaCardBuilder.ts:178]:
```suggestion
    static getHeroCard(cardText: string, buttonList: any[]): Attachment {
```

**Review by ceciliaavila — CHANGES_REQUESTED** [review]
Just some small details. Same that the .NET PR, we could add tests to cover the new code.

**Comment by anishprasad01** [comment]
@ceciliaavila 

The pipeline checks required they be generated. I simply ran the script to create them.

**Comment by ceciliaavila** [comment]
> @ceciliaavila
> 
> The pipeline checks required they be generated. I simply ran the script to create them.

The only doc requested is `botbuilder-ai.api.md` in the _etc_ folder.
I ran the checks after removing the new `doc` folder and they passed. The doc folder can be removed.

**Pull request merged**